packages/web-pkg/src/composables/piniaStores/extensionRegistry/extensionRegistry.ts

```diff
@@ -11,40 +11,45 @@ export const useExtensionRegistry = defineStore('extensionRegistry', () => {
   const registerExtensions = (e: Ref<Extension[]>) => {
     extensions.value.push(e)
   }
-  const requestExtensions = <T extends Extension>(
-    type: ExtensionType,
-    options?: {
-      scopes?: ExtensionScope[]
-      extensionPointIds?: string[]
+  const requestExtensions = <T extends Extension>(extensionPoint: ExtensionPoint<T>) => {
+    if (!extensionPoint.id || !extensionPoint.extensionType) {
+      throw new Error('ExtensionPoint must have an id and an extensionType')
     }
-  ) => {
+
     return unref(extensions).flatMap((e) =>
       unref(e).filter(
         (e) =>
-          e.type === type &&
+          e.type === extensionPoint.extensionType &&
           !configStore.options.disabledExtensions.includes(e.id) &&
-          (!options?.scopes || e.scopes?.some((s) => options?.scopes.includes(s))) &&
-          (!options?.extensionPointIds ||
-            e.extensionPointIds?.some((id) => options?.extensionPointIds.includes(id)))
+          (!e.extensionPointIds || e.extensionPointIds?.includes(extensionPoint.id))
       )
     ) as T[]
   }
 
-  const extensionPoints = ref<ExtensionPoint[]>([])
-  const registerExtensionPoint = (e: ExtensionPoint) => {
+  const extensionPoints = ref<Ref<ExtensionPoint<Extension>[]>[]>([])
+  const registerExtensionPoint = <T extends Extension>(e: ExtensionPoint<T>) => {
+    extensionPoints.value.push(ref([e]))
```

Comment:
It's somewhat fishy to create the ref on demand here - why not pass the ref as with registerExtensionPoints?

Reply:
Because when I pass a `Ref<ExtensionPoint<T>>` I need to `unref` it and then `ref` it again as array.

Reply:
Urgh..........

Reply:
Then I agree it's at least (somewhat) clear that this breaks reactivity ...

Reply:
I'm going to remove the function for registering single extension points, so that we can always require a `Ref` to be passed.